gittestserver/server.go

```diff
@@ -60,6 +61,9 @@ func NewGitServer(docroot string) *GitServer {
 	}
 }
 
+// Middleware is a git http server middleware.
+type Middleware func(http.Handler) http.Handler
```

Comment:
```suggestion
type HTTPMiddleware func(http.Handler) http.Handler
```

as we already implement both HTTP/S and SSH, it is better to be specific imo.

This would also need changing of the `AddMiddleware` method.

Reply:
Yeah, that'd be good. I was wondering if middlewares are usually used for SSH as well, but couldn't find much about it. Seems to be a thing mostly in http servers.